src/pages/home/report/EmojiPickerMenu/index.native.js

```diff
@@ -0,0 +1,119 @@
+import EmojiPickerMenuItem from '../EmojiPickerMenuItem';
+
+const propTypes = {
+    // Function to add the selected emoji to the main compose text input
+    onEmojiSelected: PropTypes.func.isRequired,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        // Ref for the emoji search input
+        this.searchInput = undefined;
+
+        // This is the number of columns in each row of the picker.
+        // Because of how flatList implements these rows, each row is an index rather than each element
+        // For this reason to make headers work, we need to have the header be the only rendered element in its row
+        // If this number is changed, emojis.js will need to be updated to have the proper number of spacer elements
+        // around each header.
+        this.numColumns = 8;
+
+        // This is the indices of each category of emojis
+        // The positions are static, and are calculated as index/numColumns (8 in our case)
+        // This is because each row of 8 emojis counts as one index
+        // If this emojis are ever added to emojis.js this will need to be updated or things will break
+        this.unfilteredHeaderIndices = [0, 34, 60, 88, 99, 121, 148];
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 300, false);
+        this.renderItem = this.renderItem.bind(this);
+
+        this.state = {
+            filteredEmojis: emojis,
+            headerIndices: this.unfilteredHeaderIndices,
+        };
+    }
+
+    /**
+     * Filter the entire list of emojis to only emojis that have the search term in their keywords
+     *
+     * @param {String} searchTerm
+     */
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            // There are no headers when searching, so we need to re-make them sticky when there is no search term
+            this.setState({filteredEmojis: emojis, headerIndices: this.unfilteredHeaderIndices});
+            return;
+        }
+        const newFilteredEmojiList = [];
+        _.each(emojis, (emoji) => {
+            if (emoji.header || emoji.code === CONST.EMOJI_SPACER) {
+                return;
+            }
+
+            if (_.find(emoji.keywords, keyword => keyword.includes(normalizedSearchTerm))) {
+                newFilteredEmojiList.push(emoji);
+            }
+        });
+
+        // Remove sticky header indices. There are no headers while searching and we don't want to make emojis sticky
+        this.setState({filteredEmojis: newFilteredEmojiList, headerIndices: []});
+    }
+
+    /**
+     * Given an emoji item object, render a component based on its type.
+     * Items with the code "SPACER" return nothing and are used to fill rows up to 8
+     * so that the sticky headers function properly
+     *
+     * @param {Object} item
+     * @returns {*}
+     */
+    renderItem({item}) {
```

Comment:
I think we could also DRY up this `renderItem` by putting it in a separate file too.

Reply:
Chatted with Rory about DRYing these up and I don't think it's necessary given that these files are going to be re-merged soon and then the code won't be duplicated anyway

Reply:
What does re-merged mean?